src/util/key.rs

```diff
@@ -80,6 +82,24 @@ pub struct PublicKey {
 }
 
 impl PublicKey {
+    /// Returns bitcoin 160-bit hash of the public key
+    pub fn pubkey_hash(&self) -> PubkeyHash {
+        if self.compressed {
+            PubkeyHash::hash(&self.key.serialize())
```

Comment:
Idem about allocation.

Reply:
Sorry, after re-reviewing this I do not see where the allocation happens. The function serializes as `[u8; cost]`, hash function takes `&[u8]` and puts it right into the engine – so we save nothing by repeating the code from inside `hash` fn.

Reply:
Yeah, it's an unfortunate name for `key.serialize()`, since most every other serialization method returns a Vec..